File: src/main/java/org/idea_sp/SourcePawnColorSettingsPage.java
package org.idea_sp;

import com.intellij.openapi.editor.colors.TextAttributesKey;
import com.intellij.openapi.fileTypes.SyntaxHighlighter;
import com.intellij.openapi.options.colors.AttributesDescriptor;
import com.intellij.openapi.options.colors.ColorDescriptor;
import com.intellij.openapi.options.colors.ColorSettingsPage;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;
 
import javax.swing.*;
import java.util.Map;
 
public class SourcePawnColorSettingsPage implements ColorSettingsPage {
    private static final AttributesDescriptor[] DESCRIPTORS = new AttributesDescriptor[]{
            new AttributesDescriptor("Operator", SourcePawnSyntaxHighlighter.OPERATOR),
            new AttributesDescriptor("Literal", SourcePawnSyntaxHighlighter.LITERAL),
            new AttributesDescriptor("Reserved", SourcePawnSyntaxHighlighter.RESERVED),
            new AttributesDescriptor("Comment", SourcePawnSyntaxHighlighter.COMMENT),
    };
 
    @Nullable
    @Override
    public Icon getIcon() {
        return SourcePawnIcons.FILE;
    }
 
    @NotNull
    @Override
    public SyntaxHighlighter getHighlighter() {
        return new SourcePawnSyntaxHighlighter();
    }
 
    @NotNull
    @Override
    public String getDemoText() {
        return "// Here's a comment!\n" +
                "static a = 4;\n" +
                "public static const char s[] = \"test\";";
    }
 
    @Nullable
    @Override
    public Map<String, TextAttributesKey> getAdditionalHighlightingTagToDescriptorMap() {
        return null;
    }
 
    @NotNull
    @Override
    public AttributesDescriptor[] getAttributeDescriptors() {
        return DESCRIPTORS;
    }
 
    @NotNull
    @Override
    public ColorDescriptor[] getColorDescriptors() {
        return ColorDescriptor.EMPTY_ARRAY;
    }
 
    @NotNull
    @Override
    public String getDisplayName() {
        return "SourcePawn";
    }
}

File: src/main/kotlin/org/idea_sp/SourcePawnColorSettingsPage.kt
package org.idea_sp

import com.intellij.openapi.editor.DefaultLanguageHighlighterColors
import com.intellij.openapi.editor.colors.TextAttributesKey
import com.intellij.openapi.fileTypes.SyntaxHighlighter
import com.intellij.openapi.options.colors.AttributesDescriptor
import com.intellij.openapi.options.colors.ColorDescriptor
import com.intellij.openapi.options.colors.ColorSettingsPage
import javax.swing.Icon

class SourcePawnColorSettingsPage : ColorSettingsPage {
    companion object {
        private val DESCRIPTORS = arrayOf(
            AttributesDescriptor(
                "Operator", TextAttributesKey.createTextAttributesKey(
                    "SOURCE_PAWN_OPERATOR",
                    DefaultLanguageHighlighterColors.OPERATION_SIGN
                )
            ),
            AttributesDescriptor(
                "Literal", TextAttributesKey.createTextAttributesKey(
                    "SOURCE_PAWN_CONSTANT",
                    DefaultLanguageHighlighterColors.CONSTANT
                )
            ),
            AttributesDescriptor(
                "Reserved", TextAttributesKey.createTextAttributesKey(
                    "SOURCE_PAWN_RESERVED",
                    DefaultLanguageHighlighterColors.KEYWORD
                )
            ),
            AttributesDescriptor(
                "Comment", TextAttributesKey.createTextAttributesKey(
                    "SOURCE_PAWN_COMMENT",
                    DefaultLanguageHighlighterColors.LINE_COMMENT
                )
            ),
        );
    }

    override fun getAttributeDescriptors(): Array<AttributesDescriptor> = DESCRIPTORS

    override fun getColorDescriptors(): Array<ColorDescriptor> = ColorDescriptor.EMPTY_ARRAY

    override fun getDisplayName(): String = "SourcePawn";

    override fun getIcon(): Icon? = SourcePawnIcons.FILE;

    override fun getHighlighter(): SyntaxHighlighter = SourcePawnSyntaxHighlighter()

    override fun getDemoText(): String = """
        // Here's a comment!
        static a = 4;
        public static const char s[] = "test";
        """.trimIndent()

    override fun getAdditionalHighlightingTagToDescriptorMap(): MutableMap<String, TextAttributesKey>? = null
}
